File: frontend/src/app/core/models/design.ts
// File: frontend/src/app/core/models/design.ts
// Version: v0.2.0
export interface Design {
  id: number;
  created_at: string;
  updated_at: string;
  sequence: string;
  sequence_len: number;
  params_json?: string | null;
  tree_json?: string | null;
  name?: string | null;
}

export interface DesignByRun extends Design {
  job_id: string;
}

File: backend/app/db/schemas/design.py
# File: backend/app/db/schemas/design.py
# Version: v0.2.0
"""Pydantic schemas for Design resources (includes GA progress)."""
from __future__ import annotations

from datetime import datetime
from pydantic import BaseModel, Field
from typing import Optional

class DesignBase(BaseModel):
    id: int
    created_at: datetime
    updated_at: datetime
    sequence: str = Field(..., description="DNA sequence (A/C/G/T/N)")
    sequence_len: int
    params_json: Optional[str] = Field(None, description="Creation parameters JSON (globals + levels)")
    tree_json: Optional[str] = Field(None, description="Serialized construction tree JSON")
    ga_progress_json: Optional[str] = Field(None, description="GA progress timeline JSON")
    name: Optional[str] = None

    class Config:
        from_attributes = True

class DesignCreateRequest(BaseModel):
    sequence: str
    params_json: Optional[str] = None
    name: Optional[str] = None

class DesignDetail(DesignBase):
    pass

class DesignByRunResponse(DesignBase):
    job_id: str
    run_note: Optional[str] = None
